src/Microsoft.AspNet.Authentication.Facebook/Notifications/FacebookAuthenticationNotifications.cs

```diff
@@ -23,16 +24,33 @@ public FacebookAuthenticationNotifications()
         /// <summary>
         /// Gets or sets the function that is invoked when the Authenticated method is invoked.
         /// </summary>
-        public Func<FacebookAuthenticatedContext, Task> OnAuthenticated { get; set; }
+        public new Func<FacebookAuthenticatedContext, Task> OnAuthenticated { get; set; }
```

Comment:
Social providers basically use a subclass of the notifications base class to implement only one notification (that replaces the generic one defined by `OAuthAuthenticationNotifications`): `OnAuthenticated`.

If we can include the access or refresh tokens directly in `OAuthAuthenticatedContext`, we can't add any provider specific property there (like `UserId`, `UserName`, `Email`) as we have no guarantee that these properties will even be returned by the user info endpoint offered by the provider.

Another approach might be to remove `Facebook/Google/MSFT/AuthenticatedContext` and the "specific" `Authenticated` notifications to keep only `OAuthAuthenticatedContext`. But in this case, users would have to deal with the raw `JToken` payload, assuming providers always return JSON payloads, which is definitely not a guarantee.